components/primitives/Form/Input.tsx

```diff
@@ -0,0 +1,48 @@
+import React from "react";
+import { cn } from "@/lib/utils/cn";
+
+interface InputProps extends React.InputHTMLAttributes<HTMLInputElement> {}
+
+const Input = React.forwardRef<HTMLInputElement, InputProps>(
+  ({ id, type, className, ...rest }, ref) => {
+    const [showPassword, setShowPassword] = React.useState(false);
+    return (
+      <div className="relative">
+        <input
+          className={cn(
+            "py-2 px-3 text-sm placeholder:text-sm block w-full border border-[#D0D5DD] bg-white rounded-md outline-0 focus:border-abeg-green-20 placeholder:text-abeg-neutral-50 text-abeg-neutral-10 disabled:bg-[#F0F2F5] disabled:border-[#D0D5DD]",
+            {
+              "pr-[32px]": type === "password",
+            },
+            className,
+          )}
+          type={type === "password" && showPassword ? "text" : type}
+          ref={ref}
+          id={id}
+          {...rest}
```

Comment:
```suggestion
          {...props}
```

please make necessary changes here too, using the term `props` is more understandable 😉